Utils/AppConstants.swift

```diff
@@ -5,86 +5,60 @@
 import UIKit
 
 public enum AppBuildChannel {
-    case Developer
-    case Aurora
-    case Fennec
     case Release
     case Beta
+    case Nightly
+    case Fennec
+    case Aurora
+    case Unknown
 }
 
 public struct AppConstants {
-
     public static let IsRunningTest = NSClassFromString("XCTestCase") != nil
 
     // True if this process is executed as part of a Fastlane Snapshot test
     public static let IsRunningFastlaneSnapshot = NSProcessInfo.processInfo().arguments.contains("FASTLANE_SNAPSHOT")
 
     /// Build Channel.
     public static let BuildChannel: AppBuildChannel = {
-#if MOZ_CHANNEL_AURORA
-    return AppBuildChannel.Aurora
-#elseif MOZ_CHANNEL_FENNEC
-    return AppBuildChannel.Fennec
-#elseif MOZ_CHANNEL_RELEASE
-    return AppBuildChannel.Release
-#elseif MOZ_CHANNEL_BETA
-    return AppBuildChannel.Beta
-#else
-    return AppBuildChannel.Developer
-#endif
+        #if MOZ_CHANNEL_RELEASE
+            return AppBuildChannel.Release
+        #elseif MOZ_CHANNEL_BETA
+            return AppBuildChannel.Beta
+        #elseif MOZ_CHANNEL_NIGHTLY
+            return AppBuildChannel.Nightly
+        #elseif MOZ_CHANNEL_FENNEC
+            return AppBuildChannel.Fennec
+        #elseif MOZ_CHANNEL_AURORA
+            return AppBuildChannel.Aurora
+        #else
+            return AppBuildChannel.Unknown
```

Comment:
Not sure if it's worth having a separate case for `Unknown` since it's something that shouldn't happen. Can we just throw an `assertionFailure` here, and/or fall back to `Fennec`?